client/client_test.go

```diff
@@ -320,6 +321,99 @@ GET /a
 	}
 }
 
+func captureUserAgent(t *testing.T) string {
+	var userAgent string
+	c, err := New(&config.Config{
+		Host:       "some",
+		Token:      "token",
+		ConfigFile: "/dev/null",
+		HTTPTransport: hc(func(r *http.Request) (*http.Response, error) {
+			// Capture the user agent via the round tripper.
+			userAgent = r.UserAgent()
+
+			return &http.Response{
+				StatusCode: 200,
+				Body:       io.NopCloser(strings.NewReader(`{}`)),
+				Request:    r,
+			}, nil
+		}),
+	})
+	require.NoError(t, err)
+
+	err = c.Do(context.Background(), "GET", "/a", nil, nil, nil)
+	require.NoError(t, err)
+
+	return userAgent
+}
+
+func TestUserAgentForDBR(t *testing.T) {
+	for _, dbrVersion := range []string{"client.0", "client.1", "15.5", "15.5.0", "13.3"} {
```

Comment:
Please include newline separators here as well for readability.

Btw, doesn't this duplicate the tests done in the `useragent` package?

Reply:
This is more robust since it does not depend upon the implementation details of the validation logic and tests it end to end. In the other test we call the `match...` function manually to verify validation does not fail.

Reply:
That's fine, but could you then also include a garbage value to test that it doesn't panic/fail.

Reply:
Added a couple of cases that need sanitization.